packages/fcl/src/wallet-provider-spec/draft.md

```diff
@@ -0,0 +1,640 @@
+
+  +-----------------+   +-----------------+   +-----------------+   +-----------------+
+  | Wallet Provider |   | Wallet Provider |   | Wallet Provider |   | Wallet Provider |
+  +-------+---------+   +-------+---------+   +-------+---------+   +-------+---------+
+          |                     |                     |                     |
+          +---------------------+--------------------------------------------
+                                |
+                        +-------+---------+                             +----------------------+
+  +-------------+   +-->| Wallet Discovery|<--+   +-------------+   +-->| Private Info Service |
+  | Application |   |   +-----------------+   |   | Application |   |   +----------------------+
+  |  (Minimal)  |   |                         |   |  (Advanced) |   |
+  | +---------+ |   |   +-----------------+   |   | +---------+ |   |   +--------------+
+  | |   FCL   |<----+-->| Profile Service |<--+---->|   FCL   |<----+-->| IPFS Service |
+  | +---------+ |       +-----------------+       | +---------+ |   |   +--------------+
+  |      ^      |                                 |      ^      |   |
+  |      |      |       +-----------------+       |      |      |   |   +----------------------+
+  |      |      |   +-->|   Access Node   |<--+   |      |      |   +-->| Notification Service |
+  |      v      |   |   +-----------------+   |   |      v      |   |   +----------------------+
+  | +---------+ |   |            ^            |   | +---------+ |   |
+  | |   SDK   |<----+            |            +---->|   SDK   | |   |   +-----------------+
+  | +---------+ |                v                | +---------+ |   +-->| Payment Service |
+  + ------------+       +-----------------+       + ------------+       +-----------------+
+                        | Flow Blockchain |
+                        +-----------------+
+```
+
+# General User Flow
+
+### User signs up or in.
+
+`fcl.authenticate()` is triggered. It looks up the **Challenge Handshake Url** with
+`config().get("challenge.handshake")`. FCL will render an iframe using the **Challenge
+Handshake Url** at the src, also passing in a couple query paramaters.
+
+> During development (and on mainnet) FCL can be configured to use the wallet directly by
+> setting the **Challenge Handshake Url** to the wallet providers **Authentication Endpoint**
+> by configuring fcl like this `config().put("challenge.handshake", "https://my-awesome-wallet-provider.com/fcl/authenticate")`.
+> On mainnet though our idea is this configuration can be removed and FCL will fallback to
+> a **Challenge Handshake** that enables the discovery of FCL compatible wallets, meaning
+> users on mainnet should be able to use the same FCL compatible wallet they trust on any
+> application that uses FCL.
+
+This iframe is responsible for the **Authentication** process, the wallet , it
```

Comment:
This reads a little awkwardly. Maybe something like:

"This iframe is responsible for the **Authentication** process. The iframe (the wallet) needs to be able to send (via 'postMessage') the users **Flow Address** and some configuration ... "

Reply:
Thanks I will have this change and some additional changes in a future PR.